### ISSUE TITLE
BAM cluster - Data Source for Automatic Migration

### ISSUE THREAD

**Issue by textanalyticsman** [oracle/weblogic-deploy-tooling#948]
Hi,

I am using WDT to create a BAM cluster and after finalizing I get this.

![image](https://user-images.githubusercontent.com/25264959/126194419-8cefbcb9-bdc8-4710-9082-727624a85a14.png)

After realizing it I have been trying to set it to None by using WDT model as is shown below.

```
        BamCluster:            
            CoherenceClusterSystemResource: SoaCoherenceCluster
            WeblogicPluginEnabled: true
            DynamicServers:
                ServerNamePrefix: 'BamCluster-'
            JTACluster:
                AbandonTimeoutSeconds: 86400
                MaxXACallMillis: 120000
                MaxTransactions: 10000
                TimeoutSeconds: 120
            DataSourceForAutomaticMigration: "None"
```
However, I am getting below error
```
WLSDPLY-12409: createDomain failed to create the domain: Failed to set attribute 
DataSourceForAutomaticMigration in path /Cluster/BamCluster to value None: Could not create generic operation:JDBCSystemResource
```
Is there a different way to set this?

TAM

**Comment by textanalyticsman** [comment]
Hello,

Could you please help us with this issue? Is it possible to open this through Oracle Support? This is an important issue for us.

Best regards,
TAM

**Comment by rjeberhard** [comment]
@alai8 or @jshum2479, are you able to help @textanalyticsman?

@textanalyticsman, yes, Oracle does provide support for all of the github.com/oracle/weblogic-* tools through your WebLogic support. The benefit is that you would have a Support engineer working on your issue; however, that engineer is likely to come and defer to the same engineers that you've already reached here.

If you have Slack, you can also reach us through the WebLogic public Slack workspace, here: https://weblogic-slack-inviter.herokuapp.com/.

**Comment by textanalyticsman** [comment]
Thanks, I will wait for your team's advise here.

**Comment by jshum2479** [comment]
The original must be set by the template, we are looking into whether WDT can unset a particular attribute, it cannot set it to none currently.

**Comment by CarolynRountree** [comment]
Can you try to delete the BAM cluster and readd?
        !BamCluster
        BamCluster:            
            CoherenceClusterSystemResource: SoaCoherenceCluster
            WeblogicPluginEnabled: true
            DynamicServers:
                ServerNamePrefix: 'BamCluster-'
            JTACluster:
                AbandonTimeoutSeconds: 86400
                MaxXACallMillis: 120000
                MaxTransactions: 10000
                TimeoutSeconds: 120

It seems the BAM template must add that cluster with the datasource migration. You cannot unset in createDomain and cannot set to None or empty string.

**Comment by textanalyticsman** [comment]
Hello,

I tried your suggestion and I got this

```
####<Jul 22, 2021 11:44:41 AM> <SEVERE> <create> <main> <WLSDPLY-12410> <createDomain failed to deploy the resources and applications to the domain: Failed to set attribute Target in path /JDBCSystemResource/BamDataSource to value BamServer01,BamServer02,BamCluster: wrong type
 @com.oracle.cie.domain.operation.config.VDomainConfig.createSimpleAspects(VDomainConfig.java:73)
oracle.weblogic.deploy.deploy.DeployException: WLSDPLY-0100: Failed to set attribute Target in path /JDBCSystemResource/BamDataSource to value BamServer01,BamServer02,BamCluster: wrong type
 @com.oracle.cie.domain.operation.config.VDomainConfig.createSimpleAspects(VDomainConfig.java:73)
        at sun.reflect.GeneratedConstructorAccessor331.newInstance(Unknown Source)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at org.python.core.PyReflectedConstructor.__call__(Unknown Source)
        at org.python.core.PyJavaInstance.__init__(Unknown Source)
        at org.python.core.PyJavaClass.__call__(PyJavaClass.java:891)
        at org.python.core.PyObject.__call__(Unknown Source)
        at wlsdeploy.exception.exception_helper$py.create_deploy_exception$5(/u01/path/weblogic-deploy/lib/python/wlsdeploy/exception/exception_helper.py:125)
        at wlsdeploy.exception.exception_helper$py.call_function(/u01/path/weblogic-deploy/lib/python/wlsdeploy/exception/exception_helper.py)
        at org.python.core.PyTableCode.call(Unknown Source)
        at org.python.core.PyTableCode.call(Unknown Source)
        at org.python.core.PyFunction.__call__(PyFunction.java:193)
        at org.python.core.PyObject._callextra(Unknown Source)
        at wlsdeploy.exception.exception_helper$py.create_exception$2(/u01/path/weblogic-deploy/lib/python/wlsdeploy/exception/exception_helper.py:71)
        at wlsdeploy.exception.exception_helper$py.call_function(/u01/path/weblogic-deploy/lib/python/wlsdeploy/exception/exception_helper.py)
```

Best regards,
TAM

**Comment by CarolynRountree** [comment]
Could you attach the createDomain.log from the weblgic-deploy/logs directory please?

**Comment by textanalyticsman** [comment]
Kindly share your email to share the log.

Thanks a lot.

TAM

**Comment by CarolynRountree** [comment]
Carolyn.Rountree@oracle.com

**Comment by CarolynRountree** [comment]
Can you run this test with the !BamCluster and see if it works for you please? 
[weblogic-deploy.zip](https://github.com/oracle/weblogic-deploy-tooling/files/6879412/weblogic-deploy.zip)

**Comment by textanalyticsman** [comment]
Hi,

I am sorry for the delay; I can see you have reproduced the issue do you still need the log file?

Thanks a lot.

Best regards,
TAM

**Comment by textanalyticsman** [comment]
I am using WDT 1.9.17 and I get this error.

```
r SoaCoherenceCluster>
    ####<Aug 25, 2021 11:27:32 AM> <INFO> <Creator> <_create_named_mbeans> <WLSDPLY-12100> <Creating Cluster with the name BamCluster>

    Issue Log for createDomain version 1.9.17 running WebLogic version 12.2.1.4.0 offline mode:

    SEVERE Messages:

            1. WLSDPLY-12409: createDomain failed to create the domain: JDBCSystemResource None not found in domain or resource group

    Total:       WARNING :     0    SEVERE :     1
  stdout_lines: <omitted>
```

This is my model

```
        BamCluster:            
            CoherenceClusterSystemResource: SoaCoherenceCluster
            WeblogicPluginEnabled: true
            DynamicServers:
                ServerNamePrefix: 'BamCluster-'
            JTACluster:
                AbandonTimeoutSeconds: 86400
                MaxXACallMillis: 120000
                MaxTransactions: 10000
                TimeoutSeconds: 120
            DataSourceForAutomaticMigration: 'None'
```

**Comment by rakillen** [comment]
hi textanalyticsman, that should be:
```
            DataSourceForAutomaticMigration: null
```
hope that helps

**Comment by textanalyticsman** [comment]
I've just sent you information requested.

**Comment by CarolynRountree** [comment]
did not receive. please check spellimg of email

**Comment by textanalyticsman** [comment]
I copied and pasted the email as has been written above, could you check your junk email?

**Comment by CarolynRountree** [comment]
mmy email provider may have blocked the type of files. Could you attach at least the log to this issue

**Comment by CarolynRountree** [comment]
This was fixed in release 1.9.17.

The solution is to set the following:
DataSourceForAutomaticMigration: null
